src/Core/Security/Authentication/Application/UseCase/Login/Login.php

```diff
@@ -0,0 +1,219 @@
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Domain\Menu\Model\Page;
+use Centreon\Domain\Repository\Interfaces\DataStorageEngineInterface;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\Application\Common\UseCase\UnauthorizedResponse;
+use Core\Security\Authentication\Application\Provider\ProviderAuthenticationFactoryInterface;
+use Core\Security\Authentication\Application\Provider\ProviderAuthenticationInterface;
+use Core\Security\Authentication\Application\Repository\ReadTokenRepositoryInterface;
+use Core\Security\Authentication\Application\Repository\WriteSessionRepositoryInterface;
+use Core\Security\Authentication\Application\Repository\WriteSessionTokenRepositoryInterface;
+use Core\Security\Authentication\Application\Repository\WriteTokenRepositoryInterface;
+use Core\Security\Authentication\Application\UseCase\LoginSession\PasswordExpiredResponse;
+use Core\Security\Authentication\Domain\Exception\AuthenticationException;
+use Core\Security\Authentication\Domain\Exception\PasswordExpiredException;
+use Core\Security\Authentication\Domain\Model\NewProviderToken;
+use Core\Security\Authentication\Domain\Model\ProviderToken;
+use Core\Security\Authentication\Infrastructure\Provider\AclUpdaterInterface;
+use Core\Security\ProviderConfiguration\Domain\Model\Provider;
+use Security\Domain\Authentication\Model\Session;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+final class Login
+{
+    use LoggerTrait;
+
+    /** @var ProviderAuthenticationInterface */
+    private ProviderAuthenticationInterface $provider;
+
+    /**
+     * @param ProviderAuthenticationFactoryInterface $providerFactory
+     * @param SessionInterface $session
+     * @param DataStorageEngineInterface $dataStorageEngine
+     * @param WriteSessionRepositoryInterface $sessionRepository
+     * @param ReadTokenRepositoryInterface $readTokenRepository
+     * @param WriteTokenRepositoryInterface $writeTokenRepository
+     * @param WriteSessionTokenRepositoryInterface $writeSessionTokenRepository
+     * @param AclUpdaterInterface $aclUpdater
+     */
+    public function __construct(
+        private ProviderAuthenticationFactoryInterface $providerFactory,
+        private SessionInterface $session,
+        private DataStorageEngineInterface $dataStorageEngine,
+        private WriteSessionRepositoryInterface $sessionRepository,
+        private ReadTokenRepositoryInterface $readTokenRepository,
+        private WriteTokenRepositoryInterface $writeTokenRepository,
+        private WriteSessionTokenRepositoryInterface $writeSessionTokenRepository,
+        private AclUpdaterInterface $aclUpdater
+    )
+    {
+    }
+
+    /**
+     * @param LoginRequest $loginRequest
+     * @param PresenterInterface $presenter
+     * @return void
+     * @throws AuthenticationException
+     */
+    public function __invoke(LoginRequest $loginRequest, PresenterInterface $presenter): void
```

Comment:
```suggestion
    public function __invoke(LoginRequest $loginRequest, LoginPresenterInterface $presenter): void
```
Following my comment on LoginController you can use here your LoginPresenterInterface